sources/api/thar-be-settings/src/service.rs

```diff
@@ -103,11 +124,37 @@ where
     Ok(service_map)
 }
 
+/// Returns string of space separated settings for given service
+fn get_changed_settings_for_service(
```

Comment:
I'm having a hard time understanding something here.  I know we only set the environment variable if a service was affected, but does that variable only include settings that are relevant to that service?  It's important because the host-containers code change includes `if !setting.contains("host-containers") { true }` which means we restart a host container if it sees any non-host-container setting.

For example, if the user changed settings.host-containers.x and settings.kubernetes.y in the same commit, and the host-containers service says it's affected by settings.host-containers, and then thar-be-settings runs, will the environment variable contain settings.kubernetes?  If so, then even host container "y" would be restarted.

Reply:
no, it wouldn't contain `settings.kubernetes` since the map would be something like `{ host-containers:  settings.host-containers.x, kubernetes: settings.kubernetes} ` so the host container service will only have the settings that affects it which is `settings.host-containers.x`

Reply:
Regrading this ```It's important because the host-containers code change includes if !setting.contains("host-containers") { true } which means we restart a host container if it sees any non-host-container setting.```
So, the settings that are sent to the host-containers are only those that affects the host-containers, say `settings.host-containers.x` or` settings.network.proxy ` The use case for the check you mentioned - `if !setting.contains("host-containers") { true }` is `settings.network`  

With `settings.kubernetes` since host-containers isn't affected by this settings, it would not be passed to the restart command of host-containers thus the check wouldn't affect them.

Reply:
Ok, I think I understand now.  This function is taking in a map where the keys are service names ("sysctl", "lockdown", "containerd"... all the "services.bla" defined in the model) and the values are the settings prefixes that the model says affect those services.  For example, the model has `services.motd`, and `metadata.settings.motd` says `affected-services = ["motd"]`, so the map would have `"motd": ["settings.motd"]` as a key/value pair.  (It could possibly have other settings prefixes in the list, too, if they also affected the motd service, but that's all we have that touches motd for now.)

So, the environment variable would only contain settings prefixes from that list, i.e. directly from our model.

I think `changed_settings` was throwing me a little bit, because they're not exactly changed settings, they're the settings metadata prefixes from the model.  I wonder if we should list the changed settings instead, since services might only care about particular keys.

Reply:
Yeah, thanks for putting it up together, that's correct. 
So, these are the `changed_settings` that user changes using apiclient, so if a user runs ```apiclient set settings.network.https-proxy="0.0.0.0:9898"   settings.host-containers.admin.control.enabled=false``` for example 

The map would be - { "host-containers" :  [ settings.host-containers.admin.source,   settings.host-containers.admin.source ]}  
which means ```settings.host-containers.admin.source and  settings.host-containers.admin.source``` are the `changed_settings` that affects host-containers and those would be passed as parameter to restart_command

Reply:
Ok, I see... the real data source for the map is the `/metadata/affected-services` API, which returns keys that are full settings names (like `settings.host-containers.admin.source`) and not the metadata prefixes from the model.

Reply:
Yeah right. It will return the affected services for `changed_settings` thus our map would be `affected_services` to `changed_settings`